frontend/fingertips-frontend/components/forms/SearchForm/searchActions.ts

```diff
@@ -0,0 +1,42 @@
+'use server';
+
+import { z } from 'zod';
+import { redirect } from 'next/navigation';
+
+const $SearchFormSchema = z.object({
+  indicator: z.string({
+    invalid_type_error: 'Please enter an indicator id',
+  }),
+});
+
+export type State = {
+  errors?: {
+    indicator?: string[];
+  };
+  message?: string | null;
+};
+
+export type SearchForm = z.infer<typeof $SearchFormSchema>;
+
+export type SearchFormState<T = SearchForm> = State & T;
+
+export async function searchIndicator(
+  prevState: SearchFormState,
+  formData: FormData
+) {
+  const validatedFields = $SearchFormSchema.safeParse({
+    indicator: formData.get('indicator'),
+  });
+
+  if (!validatedFields.success) {
+    return {
+      indicator: formData.get('indicator')?.toString() ?? '',
+      errors: validatedFields.error.flatten().fieldErrors,
+      message: 'Missing Fields',
```

Comment:
What's this return type? I think it would be good to define the return type of the function here.